src/equations/shallow_water_multilayer_1d.jl

```diff
@@ -79,19 +96,38 @@ end
 # The reference total water height H0 defaults to 0.0 but is used for the "lake-at-rest"
 # well-balancedness test cases. 
 function ShallowWaterMultiLayerEquations1D(; gravity_constant,
-                                           H0 = zero(gravity_constant), rhos)
+                                           H0 = zero(gravity_constant),
+                                           threshold_limiter = nothing,
+                                           threshold_desingularization = nothing,
+                                           threshold_partially_wet = nothing,
+                                           rhos)
 
     # Promote all variables to a common type
     _rhos = promote(rhos...)
     RealT = promote_type(eltype(_rhos), eltype(gravity_constant), eltype(H0))
     __rhos = SVector(map(RealT, _rhos))
 
+    # Set default values for thresholds
+    if threshold_limiter === nothing
+        threshold_limiter = 5 * eps(RealT)
+    end
+    if threshold_desingularization === nothing
+        threshold_desingularization = default_threshold_desingularization(RealT)
+    end
+    if threshold_partially_wet === nothing
+        threshold_partially_wet = default_threshold_partially_wet(RealT)
+    end
+
     # Extract number of layers and variables
     NLAYERS = length(rhos)
     NVARS = 2 * NLAYERS + 1
 
     return ShallowWaterMultiLayerEquations1D{NVARS, NLAYERS, RealT}(gravity_constant,
-                                                                    H0, __rhos)
+                                                                    H0,
+                                                                    threshold_limiter,
+                                                                    threshold_desingularization,
+                                                                    threshold_partially_wet,
+                                                                    __rhos)
 end
 
 @inline function Base.real(::ShallowWaterMultiLayerEquations1D{NVARS, NLAYERS, RealT}) where {
```

Comment:
In multiple parts of the code functions that return type parameters are not covered by Codecov (even though they are executed in the tests). 

```julia
@inline function Base.real(::ShallowWaterMultiLayerEquations1D{NVARS, NLAYERS, RealT}) where {
                                                                                              NVARS,
                                                                                              NLAYERS,
                                                                                              RealT <:
                                                                                              Real
                                                                                              }
    RealT
end
```

In `Trixi.jl` an analogous function is covered (https://app.codecov.io/gh/trixi-framework/Trixi.jl/pull/1824/blob/src/equations/compressible_euler_multicomponent_1d.jl#L92)

```julia
@inline function Base.real(::CompressibleEulerMulticomponentEquations1D{NVARS, NCOMP,
                                                                        RealT}) where {
                                                                                       NVARS,
                                                                                       NCOMP,
                                                                                       RealT
                                                                                       }
    RealT
end```

Any ideas why that's the case or how to fix it?

Reply:
I am not sure why these lines are not reported as being covered. Any ideas @ranocha or @sloede ?

Reply:
No idea, sorry

Reply:
The code coverage tracking stuff sometimes had some minor bugs. If you are sure that these lines are covered, don't worry about it right now and consider creating an MWE for the upstream developers.